as/plugins/org.jboss.ide.eclipse.as.classpath.core/src/org/jboss/ide/eclipse/as/classpath/core/runtime/cache/internal/ModuleAliasUtil.java

```diff
@@ -27,16 +28,25 @@ public String getAlias(File moduleXml) {
 					if( memento != null ) {
 						String nodeName = memento.getNodeName();
 						if( "module-alias".equals(nodeName)) { //$NON-NLS-1$
-							String targName = memento.getString("target-name"); //$NON-NLS-1$
-							if( targName != null )
-								return targName;
+							String targetName = memento.getString("target-name"); //$NON-NLS-1$
+							String targetSlot = memento.getString("target-slot");//$NON-NLS-1$
```

Comment:
nit picking: you could get the target slot only after checking if the name is empty, right?

Reply:
Yeah i could but i"m not gonna change this right now because then your approval will disappear ;)   I'll keep that in mind for future changes though.